var/spack/repos/builtin/packages/nlopt/package.py

```diff
@@ -32,11 +32,11 @@ class Nlopt(CMakePackage):
     other algorithms."""
 
     homepage = "https://nlopt.readthedocs.io"
-    url      = "https://github.com/stevengj/nlopt/releases/download/nlopt-2.4.2/nlopt-2.4.2.tar.gz"
+    url      = "https://github.com/stevengj/nlopt/archive/v2.5.0.tar.gz"
     git      = "https://github.com/stevengj/nlopt.git"
 
     version('develop', branch='master')
-    version('2.4.2', 'd0b8f139a4acf29b76dbae69ade8ac54')
+    version('2.5.0', 'ada08c648bf9b52faf8729412ff6dd6d')
```

Comment:
Did this work before? Generally we keep older versions around, but dealing with changed buildsystems is annoying

Reply:
From what I can gather, the initial package.py for nlopt was created to work just from the git repo, fairly recently, and this was after it had been changed to cmake. It did include the url to the 2.4.2 tar.gz, but no, it did not ever work from that. 

I have belatedly discovered an existing issue on the topic from May last year (https://github.com/spack/spack/issues/8127), which discusses the same topics, but it appears it was never resolved.